internal/lsp/server.go

```diff
@@ -683,6 +711,21 @@ func (s *Server) handleInitialized(ctx context.Context, params *lsproto.Initiali
 		NpmExecutor: s,
 		ParseCache:  s.parseCache,
 	})
+
+	if s.initializeParams != nil && s.initializeParams.InitializationOptions != nil && *s.initializeParams.InitializationOptions != nil {
+		// handle userPreferences from initializationOptions
+		userPreferences := s.session.NewUserPreferences()
+		userPreferences.Parse(*s.initializeParams.InitializationOptions)
+		s.session.InitializeWithConfig(userPreferences)
+	} else {
+		// request userPreferences if not provided at initialization
+		userPreferences, err := s.RequestConfiguration(ctx)
```

Comment:
One blocking thing here; we need to check the client capabilities to ensure that the client can even handle this request. As in `ClientCapabilities.Workspace.Configuration`.

Reply:
that's checked in `RequestConfiguration()`